include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -318,12 +330,14 @@ namespace realization {
                 forcing_params forcing_config = this->get_global_forcing_params(identifier, simulation_time_config);
 
                 std::shared_ptr<Catchment_Formulation> missing_formulation = construct_formulation(formulation_type_key, identifier, forcing_config, output_stream);
+                missing_formulation_vec.push_back(missing_formulation);
```

Comment:
I am pretty sure `missing_formulation_vec` is completely redundant with `Formulation_Manager::formulations`:
```
std::map<std::string, std::shared_ptr<Catchment_Formulation>> formulations
```
It's just a map and not a vector. You could use the contents of the map easily instead of keeping a second vector to contain them. Also this name is confusing... once they are in this vector, they are not missing anymore.

Reply:
As you can see from line 333, std::shared_ptr<Catchment_Formulation> missing_formulation is a pointer and so we are building a vector to keep all the missing_formulation in one place. Alternatively, we can use the formulations map but we have go know the catchment id when we use the map.

Reply:
The updated version in the new commit uses map. The vector is deleted.